drivers/tests/UNITTESTS/Watchdog/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set(TEST_NAME watchdog-unittest)
+
+add_executable(${TEST_NAME})
+
+target_compile_definitions(${TEST_NAME} 
+    PRIVATE
+        DEVICE_WATCHDOG
+        MBED_WDOG_ASSERT=1
+)
+
+target_sources(${TEST_NAME}
+    PRIVATE
+        ${MBED_PATH}/drivers/source/Watchdog.cpp
```

Comment:
as we are defining mbed-os project, we should use this instead https://cmake.org/cmake/help/latest/variable/PROJECT-NAME_SOURCE_DIR.html